sdk/rust/README.md

```diff
@@ -2,80 +2,107 @@
-        http::Request::builder()
-            .method("GET")
-            .uri("https://fermyon.com")
-            .body(())?,
-    ).await?;
+async fn handle_hello_world(_req: Request) -> Result<impl IntoResponse> {
+    // Create the outbound request object
+    let req = Request::builder()
+        .method(Method::Get)
+        .uri("https://random-data-api.fermyon.app/animals/json")
+        .build();
 
-    res.headers_mut()
-        .insert(http::header::SERVER, "spin/0.1.0".try_into()?);
+    // Send the request and await the response
+    let res: Response = spin_sdk::http::send(req).await?;
 
+    println!("{:?}", res);  // log the response
     Ok(res)
 }
-
 ```
 
-In order for the component above to be allowed to make the outbound HTTP
-request, the destination host must be declared in the Spin application
-configuration:
+For the component above to be allowed to make the outbound HTTP request, the destination host must be declared, using the `allowed_outbound_hosts` configuration, in the Spin application's manifest (the `spin.toml` file):
 
 ```toml
-[[component]]
-id = "hello"
-source = "target/wasm32-wasi/release/spinhelloworld.wasm"
-allowed_outbound_hosts = [ "https://fermyon.com" ]
-[component.trigger]
-route = "/hello"
+spin_manifest_version = 2
+
+[application]
+name = "hello_world"
+version = "0.1.0"
+authors = ["Your Name <your-name@example.com>"]
+description = "An example application"
+
+[[trigger.http]]
+route = "/..."
+component = "hello-world"
+
+[component.hello-world]
+source = "target/wasm32-wasi/release/hello_world.wasm"
+allowed_outbound_hosts = ["https://random-data-api.fermyon.app"]
+[component.hello-world.build]
+command = "cargo build --target wasm32-wasi --release"
+watch = ["src/**/*.rs", "Cargo.toml"]
 ```
 
-Making a request to this component, we can see the appended header, and that the
-response contains the expected body:
+### Building and Running the Spin Application
+
+Spin build can be used to build all components defined in the Spin manifest file at the same time, and also has a flag that starts the application after finishing the compilation, `spin build --up`:
+
+```bash
+$ spin build --up
+Building component hello-world with `cargo build --target wasm32-wasi --release`
+    Finished release [optimized] target(s) in 0.12s
+Finished building all Spin components
+Logging component stdio to ".spin/logs/"
 
+Serving http://127.0.0.1:3000
+Available Routes:
+  hello-world: http://127.0.0.1:3000 (wildcard)
 ```
-$ curl -I localhost:3000/hello
+
+Once our application is running, we can make a request (by visiting `http://localhost:3000/` in a web browser) or using `curl` as shown below:
+
+```bash
+$ curl -i localhost:3000
 HTTP/1.1 200 OK
-content-length: 29350
-content-type: text/html; charset=utf-8
-server: spin/0.1.0
+content-length: 77
+content-type: application/json
+
+{"timestamp":1702599575198,"fact":"Sharks lay the biggest eggs in the world"}
 ```
```

Comment:
```suggestion
```

## Fermyon Developer Home

For many more examples, please see the official [Fermyon Developer Home](https://developer.fermyon.com/). 

[The language support overview page](https://developer.fermyon.com/spin/v2/language-support-overview) lists all of Spin's features (key-value storage, SQLite, MySQL, Redis, Serverless AI etc.) and links to how each feature is implemented in a given language (Rust, TS/JS, Python, TinyGo).
```